src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -1104,11 +1104,15 @@ public void writeBytes(byte[] data, int off, int len) throws IOException
 
     // Stream Terminators
 
-    public void flush() throws IOException
+    public void flush()
     {
         if (getDepth() == 0 && !user.hasAnnotations() && (localsLocked || lstAppendEnabled))
         {
-            unsafeFlush();
+            try {
+                unsafeFlush();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
We should not change the exception throwing behavior here. We may need to define our own Functional Interface, like

```
@FunctionalInterface
class ThrowingRunnable() {
    void run() throws IOException;
}
```

`this::flush()` should be able to conform to that without changing its signature.